test/integration/integration_test.go

```diff
@@ -214,34 +243,52 @@ func (i *integrationTest) exerciseCleanBuild(tag string, verifyCallback bool, im
 		t.Fatalf("Sti build did not invoke callback with valid json message")
 	}
 
-	i.checkForImage(tag)
-	containerID := i.createContainer(tag)
-	defer i.removeContainer(containerID)
-	i.checkBasicBuildState(containerID, resp.WorkingDir)
+	if expectImageName {
```

Comment:
you can also revert the logic and check `if ! expectedImageName`. In that case just move the `if setTag` there and return. 

Then you just get one less if. Then `Fatalf` actually 'exit' so there is no need for `else` there. So this can look like this:

```go
if setTag {
  # ...
}

if ! expectedImageName && len(resp.ImageID) > 0 {
  t.Fatalf("...")
}

if len(resp.ImageID) == 0 {
  t.Fatalf("...")
}

```